src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -39,10 +39,18 @@ public DiagnosticState GetState(StateType stateType)
                 return _state[(int)stateType];
             }
 
-            public void Remove(object documentOrProjectId)
+            public void Remove(object documentOrProjectId, bool includeProjectState = true)
```

Comment:
Probably easier to understand if the flag is called "onlyDocumentStates" and the check below is negated.